### PR TITLE
Removed all the generics in classes that doesn't really need to be generic

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Trying to make code cleaner and easier to read

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### CODE REVIEW

File: data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java [hunk 1]
```diff
@@ -59,7 +58,8 @@
   // dot notation names to the monitoring system’s recommended naming convention.
   // Additionally, this naming convention implementation sanitizes metric names and tags of special characters that
   // are disallowed by the monitoring system.
-  public static final String HTTP_EVENTS_SENT_COUNT = "http.events.sent"; // prometheus format --> http_events_sent_total
+  public static final String HTTP_EVENTS_SENT_COUNT = "http.events.sent";
+  // prometheus format --> http_events_sent_total
```
Comment (pierDipi) [inline data-plane/dispatcher/src/main/java/dev/knative/eventing/kafka/broker/dispatcher/Main.java:62]: Nit: swap these 2 lines.

File: data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java [hunk 2]
```diff
@@ -56,8 +56,10 @@
   // dot notation names to the monitoring system’s recommended naming convention.
   // Additionally, this naming convention implementation sanitizes metric names and tags of special characters that
   // are disallowed by the monitoring system.
-  public static final String HTTP_REQUESTS_MALFORMED_COUNT = "http.requests.malformed"; // prometheus format --> http_requests_malformed_total
-  public static final String HTTP_REQUESTS_PRODUCE_COUNT = "http.requests.produce";     // prometheus format --> http_requests_produce_total
+  public static final String HTTP_REQUESTS_MALFORMED_COUNT = "http.requests.malformed";
+  // prometheus format --> http_requests_malformed_total
+  public static final String HTTP_REQUESTS_PRODUCE_COUNT = "http.requests.produce";
+  // prometheus format --> http_requests_produce_total
```
Comment (pierDipi) [inline data-plane/receiver/src/main/java/dev/knative/eventing/kafka/broker/receiver/Main.java:62]: Same here to both.

**Comment by slinkydeveloper** [comment]
@pierDipi Any idea about https://prow.knative.dev/view/gs/knative-prow/pr-logs/pull/knative-sandbox_eventing-kafka-broker/566/pull-knative-sandbox-eventing-kafka-broker-unit-tests/1354724496065433600#1:build-log.txt%3A2228?

File: data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/utils/ReferenceCounter.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
+package dev.knative.eventing.kafka.broker.core.utils;
```
Comment (pierDipi) [inline data-plane/core/src/main/java/dev/knative/eventing/kafka/broker/core/utils/ReferenceCounter.java:1]: `Some files do not have the expected license header`

**Review by pierDipi — APPROVED** [review]
/lgtm
/approve

**Pull request merged**